lib/dockerUtil.js

```diff
@@ -0,0 +1,227 @@
+const ERR = require('async-stacktrace');
+const AWS = require('aws-sdk');
+const Docker = require('dockerode');
+const _ = require('lodash');
+const moment = require('moment');
+const debug = require('debug')('prairielearn:dockerUtil');
+
+const config = require('./config');
+const logger = require('./logger');
+
+var dockerUtil = {};
+var docker = new Docker();
+
+// Storing right now as a local variable
+// TODO: might want to put in lib/cache to share and handle expiration
+var dockerAuthData = null;
+
+function AuthDataExtractLogin(data) {
+    let buff = Buffer.from(data.authorizationToken, 'base64');
+    let authString = buff.toString('ascii');
+    let authArray = authString.split(':');
+
+    var auth = {
+    username: authArray[0],
+    password: authArray[1],
+    };
+    debug(auth);
+    return auth;
+}
+
+dockerUtil.setupDockerAuth = function(callback) {
+
+    // If we have cached data that's not within an hour of expiring, use it
+    if (dockerAuthData && moment().isBefore(moment(dockerAuthData.expiresAt).subtract(1, 'hour'))) {
+        debug(dockerAuthData);
+        debug('Using cached ECR authorization token');
```

Comment:
```suggestion
        logger.info('Using cached ECR authorization token');
```

To match the `info` logging below when we get the token.